spec/lib/cyclonedx/report_ruby_gems_cyclonedx_spec.rb

```diff
@@ -0,0 +1,155 @@
+require_relative '../../spec_helper'
+require 'json'
+
+describe Cyclonedx::ReportRubyGems do
+  describe "#run" do
+    it 'should report all the deps in the Gemfile if Gemfile.lock is absent in cyclonedx' do
+      repo = Salus::Repo.new('spec/fixtures/report_ruby_gems/gemfile_only')
+      scanner = Salus::Scanners::ReportRubyGems.new(repository: repo, config: {})
+      scanner.run
+
+      ruby_cyclonedx = Cyclonedx::ReportRubyGems.new(scanner.report)
+      expect(ruby_cyclonedx.build_components_object).to match_array(
+        [
+          {
+            "bom-ref": "pkg:gem/kibana_url",
+            "type": "library",
+            "group": "",
+            "name": "kibana_url",
+            "version": "",
+            "purl": "pkg:gem/kibana_url",
+            "properties": [
+              {
+                "key": "source",
+                "value": "https://rubygems.org/"
```

Comment:
Seems like all of your test cases use `https://rubygems.org` as the only gem source.
Some Gemfile.locks have multiple sources.  Have you tested with these?  
This would be good as an additional test case.

Reply:
Good idea, will add!